### ISSUE TITLE
Nightly test failure, Hip, Sycl backends, SortByKey failure

### ISSUE THREAD

**Issue by ndellingwood** [kokkos/kokkos#6848]
**Describe the bug**

Nightly Hip builds with rocm/5.6.0 on MI250 are failing in the `hip.SortByKey` subtest:
```
21:05:26 [ RUN      ] hip.SortByKey
21:05:26 Kokkos::View ERROR: attempt to access inaccessible memory space (label="")
21:05:26 Backtrace:
21:05:26 [0x642c00] 
21:05:26 [0x619632] 
21:05:26 [0x46b3c6] 
21:05:26 [0x4c5dfd] 
21:05:26 [0x4c70fc] 
21:05:26 [0x4c3d83] 
21:05:26 [0x4c32b5] 
21:05:26 [0x4c2eac] 
21:05:26 [0x4c2baf] 
21:05:26 [0x4c1b0b] 
21:05:26 [0x4a7aa9] 
21:05:26 [0x60f7e7] 
21:05:26 [0x5e9d30] 
21:05:26 [0x5eb060] 
21:05:26 [0x5eb912] 
21:05:26 [0x5fb58d] 
21:05:26 [0x6104d7] 
21:05:26 [0x5fb04d] 
21:05:26 [0x462150] 
21:05:26 [0x7f47757e7d85] __libc_start_main
21:05:26 [0x46207e] 
```

This likely follows after merge of #6801 

**Please include the following for a minimal reproducer**

**Reproducer** (caraway MI250 queue):
```bash
# salloc a node, specific to caraway
salloc -N 1 -p MI250

# env
module load cmake rocm/5.6.0

# configure
${KOKKOS_PATH}/generate_makefile.bash --with-hip --with-serial --compiler=hipcc --arch=VEGA90A --kokkos-path=${KOKKOS_PATH}

# build
make -j16

# test
ctest -R Kokkos_UnitTest_Sort
```

**Comment by dalg24** [comment]
The issue is there
https://github.com/kokkos/kokkos/pull/6801/files#r1507542399

**Comment by ndellingwood** [comment]
The Hip failures were resolved after merge of #6856 , thanks!
